jphp-core/src/org/develnext/jphp/core/compiler/jvm/statement/expr/ForCompiler.java

```diff
@@ -37,11 +39,18 @@ public void write(ForStmtToken token) {
         LabelNode iter = new LabelNode();
         LabelNode end = new LabelNode();
 
-        expr.writeExpression(token.getCondition(), true, false);
-        expr.writePopBoolean();
+	for (Iterator<ExprStmtToken> i = token.getConditionExpr().iterator(); i.hasNext();) {
+	    ExprStmtToken expr = i.next();
+	    if (i.hasNext()) {
+		this.expr.writeExpression(expr, false, false);
+	    } else {
+		this.expr.writeExpression(expr, true, false);
+		this.expr.writePopBoolean();
```

Comment:
> Is only the last expression will be checked?

Yes.

> Each of the expressions can be empty or contain multiple expressions separated by commas. In expr2, all expressions separated by a comma are evaluated but the result is taken from the last part.

(via http://php.net/manual/en/control-structures.for.php)

```php
for ($i = 0, $x = 0; $i < 5, $x < 10; $i++, $x++) {
     echo "$i, $x\n";     
}
```

This code outputs `0, 0...9, 9` on original PHP. I checked with PHP 5.3.27, 5.4.30, 5.5.14, 5.6.3.